File: sfas-ui/src/app/core/services/auth.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { tap, Observable } from 'rxjs';
import { AuthStore } from '../stores/auth.store';
import { AuthResponse } from '../models/user.model';
import { environment } from '../../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class AuthService {
  private http = inject(HttpClient);
  private authStore = inject(AuthStore);
  private baseUrl = environment.apiUrl + '/api/v1/auth';

  login(email: string, password: string): Observable<AuthResponse> {
    return this.http.post<AuthResponse>(`${this.baseUrl}/login`, { email, password }).pipe(
      tap(response => {
        const user = {
          id: response.userId,
          name: response.name,
          email: response.email,
          role: response.role
        };
        this.authStore.setAuth(response.token, user);
      })
    );
  }

  logout(): void {
    this.authStore.clearAuth();
  }
}

File: sfas_mobile/lib/features/auth/services/auth_service.dart
import 'package:dio/dio.dart';
import '../../../core/constants/endpoints.dart';
import '../models/login_response.dart';

class AuthService {
  final Dio _dio;

  AuthService(this._dio);

  Future<LoginResponse> login(String email, String password) async {
    try {
      final response = await _dio.post(
        Endpoints.login,
        data: {
          'email': email,
          'password': password,
        },
      );
      
      return LoginResponse.fromJson(response.data);
    } catch (e) {
      rethrow;
    }
  }
}
